### ISSUE TITLE
wc_set_time_limit impacts PHP time limit for CLI import

### ISSUE THREAD

**Issue by jonny-bull** [woocommerce/woocommerce-gateway-stripe#1527]
**Describe the bug**

wc_set_time_limit impacts PHP execution time outside of actions performed by the Stripe gateway plugin.

On running an import via CLI, the command failed consistently after a minute and a half, complaining about a 70 second maximum execution time (see screenshot below).

This was on my local environment, which has max_execution_time set to 0 to prevent timeouts. 

[This Stack Overflow question](https://stackoverflow.com/questions/66647956/woocommerce-php-api-timeout-70-seconds-despite-max-execution-time-600-seconds) was the only thing that pointed me in the right direction.

Commenting out the following line allowed the command to run unimpeded:
https://github.com/woocommerce/woocommerce-gateway-stripe/blob/64afc2727f3b6ce1b4dc644f4f526ebc99310ef1/includes/connect/class-wc-stripe-connect-api.php#L116

**To Reproduce**
1. Export a large site to WordPress XML (my export file was 38Mb)
2. Import using the CLI `wp import [FILE NAME].xml --authors=create`
3. Wait about a minute and a half

**Expected behavior**
Expected the import command to run unimpeded by PHP execution timeouts

**Screenshots**

![Screenshot 2021-04-12 at 14 16 35](https://user-images.githubusercontent.com/53259768/114417868-e47b5b80-9ba9-11eb-93a4-e5db9a8f485d.png)

**Environment (please complete the following information):**
 - WordPress Version 5.5.1
 - WooCommerce Version 5.1.0
 - Stripe Plugin Version 5.0.0
 - Browser N/A

This is a network site

**Comment by lkraav** [comment]
Indeed, these sorts of things definitely need to verify whether request context is theirs to modify.
